tests/unit/test_label_roundtrip.py

```
@@ -0,0 +1,166 @@
+        [
+            {"cls": 0, "cx": 1.4, "cy": -0.2, "w": 0.5, "h": 0.5},  # clamp
+            {"cls": 9, "cx": 0.5, "cy": 0.5, "w": 0.1, "h": 0.1},  # cls >= nc -> drop
+            {"cls": 0, "cx": 0.5, "cy": 0.5, "w": 0.0, "h": 0.1},  # zero area -> drop
+        ],
+        nc=2,
+    )
+    assert len(out) == 1
+    assert out[0]["cx"] == 1.0 and out[0]["cy"] == 0.0
+
+
+def _make_dataset(root, with_images_in_root=False):
```

Ruff (0.15.15)

[warning] Missing return type annotation for private function `_make_dataset`

(ANN202)